core/src/main/java/io/confluent/rest/validation/JacksonMessageBodyProvider.java

```diff
@@ -77,6 +82,8 @@ public Object readFrom(Class<Object> type,
       Throwable cause = e.getCause();
       if (cause instanceof ConstraintViolationException) {
         throw (ConstraintViolationException) cause;
+      }else if(cause instanceof WebApplicationException){
```

Comment:
Sorry I mis-understood your comment. The online Jackson doc specified that FAIL_ON_READING_DUP_TREE_KEY may trigger the JsonMappingException only. They did not mention specific cause type.